plugins/modules/source_control/gitlab/gitlab_service.py

```diff
@@ -0,0 +1,468 @@
+            setattr(obj, k, v)
+
+
+def main():
+    definitions = init_definitions()
+
+    base_spec = basic_auth_argument_spec()
+    base_spec.update(dict(
+        api_token=dict(type='str', no_log=True),
+        project=dict(required=True),
+        service=dict(required=True, type='str', choices=list(definitions.keys())),
+        active=dict(required=False, default=True, type='bool'),
+        params=dict(required=False, type='dict'),
+        events=dict(required=False, type='list', elements='str', default=GitLab_Services.HOOK_EVENTS, choices=GitLab_Services.HOOK_EVENTS),
+        state=dict(default='present', choices=['present', 'absent']),
+    ))
+
+    constraints = dict(
+        mutually_exclusive=[
+            ['api_username', 'api_token'],
+            ['api_password', 'api_token']
+        ],
+        required_together=[
+            ['api_username', 'api_password']
+        ],
+        required_one_of=[
+            ['api_username', 'api_token']
+        ],
+        required_if=[
+            ['state', 'present', ['params']]
+        ],
+        supports_check_mode=True
+    )
+
+    stub_init = AnsibleModule(argument_spec=base_spec, **constraints)
+    service = stub_init.params['service']
+    state = stub_init.params['state']
+    if state == 'present':
+        # since we know the service (which has been validated), recreate a module_specs to validate suboptions
+        # sub_arg_specs = {i: definitions[service][i] for i in definitions[service] if i != '_events'}
+        # python2.6 remove-me:
+        sub_arg_specs = {}
+        for i in iteritems(definitions[service]):
+            if i != '_events':
+                sub_arg_specs[i] = definitions[service][i]
+
+        base_spec['params'] = dict(required=False, type='dict', options=sub_arg_specs)
+        if '_events' in definitions[service]:
+            base_spec['events'] = dict(required=True, type='list', elements='str', choices=definitions[service]['_events'])
+        module = AnsibleModule(argument_spec=base_spec, **constraints)
+    else:
+        module = stub_init
+
+    project = module.params['project']
+    active = True
+    # active = module.params['active']
+    params = module.params['params']
+    events = module.params['events']
+
+    if not HAS_GITLAB_PACKAGE:
+        module.fail_json(msg=missing_required_lib("python-gitlab"), exception=GITLAB_IMP_ERR)
+
+    gitlab_instance = gitlabAuthentication(module)
+
+    try:
+        project = gitlab_instance.projects.get(project)
+    except gitlab.GitlabGetError as e:
+        module.fail_json(msg='No such a project %s' % project, exception=to_native(e))
+
+    try:
+        remote_service = project.services.get(service)
+        original_attributes = remote_service.attributes.copy()
+    except gitlab.GitlabGetError as e:
+        module.fail_json(msg='No such a service %s' % service, exception=to_native(e))
+
+    Services_Helper = GitLab_Services(module, service)
```

Comment:
Naming convention for variables is lower-case:
```suggestion
    services_helper = GitLab_Services(module, service)
```